### DISCUSSION TITLE
gatsby-plugin-google-gtag window.gtag is not a function error in local dev

### DISCUSSION THREAD

**Discussion by edmanolache** [gatsbyjs/gatsby#37973] [Help]
How can I ignore the window.gtag is not a function error in local development while running gatsby develop? 

I have custom GA4 events that I want to send using the window.gtag() function which works fine in production but in development mode I get an unhandled runtime error.

**Comment by edmanolache** [comment]
For now, I'm wrapping my window.gtag() calls in an if statement that checks if process.env.NODE_ENV !== "development" but I still want to know if there's a better solution.

**Reply by gh640** [reply]
It'd be good to use `process.env.NODE_ENV`. `gatsgby-plugin-google-gtag` uses the following condition:

```js
process.env.NODE_ENV !== `production` || typeof gtag !== `function`
```

`gatsby-browser.js`:

https://github.com/gatsbyjs/gatsby/blob/6feee5d6bb752f458f01d968798dad18e5b43d30/packages/gatsby-plugin-google-gtag/src/gatsby-browser.js#L1-L4

`gatsby-ssr.js`:

https://github.com/gatsbyjs/gatsby/blob/6feee5d6bb752f458f01d968798dad18e5b43d30/packages/gatsby-plugin-google-gtag/src/gatsby-ssr.js#L4-L9
